### ISSUE TITLE
Redundant `to_string`?

### ISSUE THREAD

**Issue by smoelius** [rust-lang/rust-clippy#7729]
I was surprised that there does not seem to be a lint to catch the following. Is it possible that it exists and I am somehow not triggering it?
```rust
use camino::Utf8PathBuf;

fn main() {
    let path = Utf8PathBuf::from("src/main.rs");
    println!("{}", path.to_string()); // to_string is unnecessary since Utf8PathBuf implements Display
    println!("{}", path);
}
```

**Comment by camsteffen** [comment]
Related to #7729. I'd call this `to_string_in_format_args`.

**Comment by akanalytics** [comment]
Is it true that 
a) println!("{:<20}", struct.to_string()); 
gives the same result as 
b) println!("{:<20}", struct); 

?

Doesnt it depend on how the formatter is written. 
Line a) uses the default  "{}" display to render a string then renders the string using the formatting rules :<20
Whereas b) uses the formatting rules of the struct's display, not the string's display trait impl.

**Comment by camsteffen** [comment]
@akanalytics Feel free to file a bug if you found a false positive.
